src/UI/templates/js/MainControls/src/mainbar.main.js

```diff
@@ -270,7 +270,9 @@ var mainbar = function() {
             }
         }
         mb.model.actions.initMoreButton(mb.renderer.calcAmountOfButtons());
-        mb.renderer.render(mb.model.getState());
+        let state = mb.model.getState();
+        mb.persistence.storePageState(state.any_entry_engaged || state.tools_engaged);
```

Comment:
As from my question below: can we use `mb.persistence.store(state)` here instead?

Reply:
Hey @thibsy ,
yes. In fact, I had it like this before, but in store(state), all entries are being walked through to compress them. 
This is of no concern to determine the slate-state, and I wanted max speed, since the whole thing is about "twiching".
Furthermore, I see no real disadvantage in exposing the function - closing the slate or setting the cookie on click of the button would actually have prevented the remarked behavior, and - it's a subset of a previously exposed function, without  there being any relevant calculations beforehand (in storeStates).
Best regards,
Nils

Reply:
Thx @nhaagen for the explanation. I hesitated because as I understand the mainbar js is written in a [redux-ish](https://redux.js.org/introduction/core-concepts) manner, which tries to avoid storing state "arbitrarily" and only does so through actions. As far as I understand, you always want to store the entire state, not just parts of it. I agree that speed is of the essence here, but is it really `.store()` vs `.storePageState()` that matters? IMO we could accept this, but I'd ask for a brief comment in `storePageState` explaining the exposure.